crates/astria-conductor/src/conductor.rs

```diff
@@ -77,20 +87,145 @@ impl Future for Handle {
     }
 }
 
+/// A wrapper around [`InnerConductorTask`] that manages shutdown and restart of the conductor.
 pub struct Conductor {
     /// Token to signal to all tasks to shut down gracefully.
     shutdown: CancellationToken,
 
+    /// Token to signal to inner conductor task to shut down gracefully.
+    conductor_inner_shutdown: CancellationToken,
+
+    /// Handle for the inner conductor task.
+    handle: ConductorInnerHandle,
+
+    /// Configuration for the conductor, necessary upon a restart.
+    cfg: Config,
+
+    /// Metrics used by tasks, necessary upon a restart.
+    metrics: &'static Metrics,
+}
+
+impl Conductor {
+    /// Creates a new `Conductor` from a [`Config`].
+    ///
+    /// # Errors
+    /// Returns an error if [`InnerConductorTask`] could not be created.
+    pub fn new(cfg: Config, metrics: &'static Metrics) -> eyre::Result<Self> {
+        let conductor = InnerConductorTask::new(cfg.clone(), metrics)?;
+        let conductor_inner_shutdown = conductor.shutdown.clone();
```

Comment:
As noted above, this is redundant. It's better to create a child token and pass that into the inner conductor.